services/proxy/pkg/middleware/security.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"github.com/owncloud/ocis/v2/services/proxy/pkg/config"
+	"github.com/unrolled/secure"
+	"github.com/unrolled/secure/cspbuilder"
+	"gopkg.in/yaml.v2"
+	"net/http"
+	"os"
+)
+
+func LoadCSPConfig(proxyCfg *config.Config) (*config.CSP, error) {
+	cspConfig := config.CSP{}
+	yamlContent, err := loadCSPYaml(proxyCfg)
+	if err != nil {
+		return nil, err
+	}
+	err = yaml.Unmarshal(yamlContent, &cspConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cspConfig, nil
+}
+
+func loadCSPYaml(proxyCfg *config.Config) ([]byte, error) {
+	if proxyCfg.CSPConfigFileLocation == nil {
+		return []byte(config.DefaultCSPConfig), nil
+	}
+	yamlContent, err := os.ReadFile(*proxyCfg.CSPConfigFileLocation)
+	if err != nil {
+		return nil, err
+	}
+	return yamlContent, nil
```

Comment:
Maybe use shorter
```suggestion
	return os.ReadFile(*proxyCfg.CSPConfigFileLocation)
```

We are not doing anything with the error, so why check it?